docs/src/man/basics.md

```diff
@@ -1510,32 +1510,96 @@ julia> german[Not(5), r"S"]
-- `source_column` just keep the source column as is in the result without any transformation;
-
-These rules are typically called transformation mini-language.
-
+In DataFrames.jl there are seven functions that can be used to transform data frame columns:
+
+| Function     | Memory Usage                     | Column Retention                             | Row Retention                                       |
+| ------------ | -------------------------------- | -------------------------------------------- | --------------------------------------------------- |
+| `transform`  | Creates a new data frame.        | Retains both source and transformed columns. | Retains same number of rows as source data frame.   |
+| `transform!` | Modifies an existing data frame. | Retains both source and transformed columns. | Retains same number of rows as source data frame.   |
+| `select`     | Creates a new data frame.        | Retains only transformed columns.            | Retains same number of rows as source data frame.   |
+| `select!`    | Modifies an existing data frame. | Retains only transformed columns.            | Retains same number of rows as source data frame.   |
+| `subset`     | Creates a new data frame.        | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `subset!`    | Modifies an existing data frame. | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `combine`    | Creates a new data frame.        | Retains only transformed columns.            | Number of rows is determined by the transformation. |
+
+### Constructing Transformation Pairs
+All of the functions above use the same syntax which is commonly
+`transformation_function(source_dataframe, transformation)`.
+The `transformation` argument is a `Pair` which defines the
+transformation to be applied to the `source_dataframe`,
+and it can take any of the forms listed below:
+
+- `source_column_selector => function`
+  - passes source column(s) to function
+  - automatically names the resulting column(s)
+- `source_column_selector => function => new_column_names`
+  - passes source column(s) to function
+  - names the resulting column(s) `new_column_names`
+    - *cannot be used with `subset` or `subset!`*
+- `source_column_selector => new_column_names`
+  - renames source column
+    - *may only select one column unless broadcasting*
+    - *may be used to split a column containing collections into multiple columns*
+    - *cannot be used with `subset` or `subset!`*
+- `source_column_selector`
+  - selects source column(s) without transforming them
+    - *often used with `select` or `select!` for isolating or moving columns*
+
+#### `source_column_selector`
+The most basic `source_column_selector` is a column name,
+but there are many more ways to select columns as explained
+in the first section of the [Indexing](@ref) API.
```

Comment:
A special case of `source_column_selector` is `AsTable` that should be mentioned somewhere.

Reply:
I ignored the existence of `AsTable` here.

1. I've never used it.
2. I don't totally understand it.
3. It seems to break most of the rules I've neatly outlined.
4. Probably not for beginners?

Might be better to write a separate section on it.

Reply:
> Might be better to write a separate section on it.

This is OK (and better to be done at the end)

> It seems to break most of the rules I've neatly outlined.

The rule is simple - instead of passing columns as consecutive positional argument you pass a `NamedTuple` as a single argument.

> Probably not for beginners?

Even beginners want to do row aggregation often like:
```
julia> df = DataFrame(ones(1, 100), :auto)
1×100 DataFrame
 Row │ x1       x2       x3       x4       x5       x6       x7       x8       x9       x10      x11      x12      x13      x14      x15      x16      x17    ⋯
     │ Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float64  Float6 ⋯
─────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │     1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1.0      1. ⋯
                                                                                                                                             84 columns omitted

julia> select(df, AsTable(:) => sum)
1×1 DataFrame
 Row │ x1_x2_etc_sum
     │ Float64
─────┼───────────────
   1 │         100.0
```
I think this example or similar with `mean` should be included somewhere as it is super common.

In general use `AsTable` if you have a function that *accepts a collection as its input* as opposed to list of arguments.

Note that these two operations are equivalent (also might be included to show the difference):

```
julia> df = DataFrame([1 2 3 4], :auto)
1×4 DataFrame
 Row │ x1     x2     x3     x4
     │ Int64  Int64  Int64  Int64
─────┼────────────────────────────
   1 │     1      2      3      4

julia> select(df, r"x" => ByRow(min))
1×1 DataFrame
 Row │ x1_x2_etc_min
     │ Int64
─────┼───────────────
   1 │             1

julia> select(df, AsTable(r"x") => ByRow(minimum))
1×1 DataFrame
 Row │ x1_x2_etc_minimum
     │ Int64
─────┼───────────────────
   1 │                 1
```
If someone understands the difference between `min` and `minimum` then the need for `AsTable` will be clear after seeing this example.

Reply:
Of course there are more advanced uses of `AsTable` (where you would refer to the actual column names as you can extract them out from a `NamedTuple`, but the ones I have listed above are most common).